src/subdomains/core/buy-crypto/process/entities/buy-crypto.entity.ts

```diff
@@ -365,6 +365,7 @@ export class BuyCrypto extends IEntity {
     minFeeAmountFiat: number,
     totalFeeAmountChf: number,
     feeConstraints: BuyCryptoFee,
+    priceSteps: PriceStep[],
   ): UpdateResult<BuyCrypto> {
```

Comment:
I think the price steps should be set during output amount calculation `calculateOutputReferenceAmount`